src/ui/src/builder/useSocketTimeout.ts

```diff
@@ -43,34 +66,36 @@ export function useSocketTimeout(wf: Core, timeoutMin: number) {
 			socketClosed.value = false;
 			logger.info(`[SocketTimeout] Socket reconnected successfully`);
 		} catch (error) {
-			logger.error(`[SocketTimeout] Failed to reconnect socket, reloading page`, error);
+			logger.error(
+				`[SocketTimeout] Failed to reconnect socket, reloading page`,
+				error,
+			);
 			window.location.reload(); // fallback to full reload
 		} finally {
 			reconnecting.value = false;
 		}
 	}
 
-	const abort = useAbortController();
+	function onVisibilityChange() {
+		if (document.visibilityState === "visible") {
+			clearSchedule();
+		} else if (canCloseSocket.value) {
+			schedule();
+		}
+	}
 
 	onMounted(() => {
-		document.addEventListener(
-			"visibilitychange",
-			() => {
-				if (document.visibilityState === "visible") {
-					clearSchedule();
-				} else if (!prevent.value) {
-					schedule();
-				}
-			},
-			{ signal: abort.signal },
-		);
+		document.addEventListener("visibilitychange", onVisibilityChange, {
```

Comment:
don't we need to remove listener when component is unmounted?

Reply:
We use a signal, which means the listener will be closed when the `AbortController` is aborted 

https://github.com/writer/writer-framework/blob/655fc69622f39914e6c22d417f9b56eab6d10b97/src/ui/src/builder/useSocketTimeout.ts#L89-L89